edgePy/DGEList.py

```diff
@@ -272,37 +307,144 @@ def library_size(self) -> np.array:
         """
         return np.sum(self.counts, 0)
 
-    def cpm(self, log: bool = False, prior_count: float = PRIOR_COUNT) -> None:
-        """Return the DGEList normalized to read counts per million."""
-        self.counts = 1e6 * self.counts / np.sum(self.counts, axis=0)
-        if log:
-            self.counts[self.counts == 0] = prior_count
-            self.counts = np.log(self.counts)
+    def log_transform(self, counts, prior_count):
+        """Compute the log of the counts"""
+        counts[counts == 0] = prior_count
+        return np.log(counts)
+
+    def cpm(self, transform_to_log: bool = False, prior_count: float = PRIOR_COUNT) -> "DGEList":
+        """Normalize the DGEList to read counts per million."""
+        counts = 1e6 * self.counts / np.sum(self.counts, axis=0)
+        current_log = self.current_log_status
+        if transform_to_log:
+            counts = self.log_transform(counts, prior_count)
+            current_log = True
+
+        return self.copy(counts=counts, current_log=current_log)
 
     def rpkm(
-        self, gene_lengths: Mapping, log: bool = False, prior_count: float = PRIOR_COUNT
-    ) -> None:
+        self,
+        gene_data: CanonicalDataStore,
```

Comment:
This one still forces users to use our the CanonicalDataStore. Imagine a user already has an array of gene/feature lengths. To use his own array he would need to convert it to the data structure used by CanonicalDataStore and then create a CanonicalDataStore out of it.
One way to provide a convenient interface would be this:  

- gene_data: CanonicalDataStore: Union[CanonicalDataStore, np.array]

So users can either use the convenient CanonicalDataStore, or provide their own feature lengths (if they have special needs).

Reply:
Yes, that's a good point.... I'll think about this, to see if I can come up with an easy way to make this work.  Off hand, I'm not seeing it, since you'd have to work really hard to provide gene lengths in an alternate format, plus validation.

Reply:
What's wrong with a simple 1D array (every gene has a length)?

Reply:
Issue filed